integration-tests/tests/src/hooks/open-realm-before.ts

```diff
@@ -46,6 +46,9 @@ export function openRealmHook(config: LocalConfiguration | SyncedConfiguration =
       } as Realm.Configuration;
       this.realm = new Realm(this.config);
       // Upload the schema, ensuring a valid connection
+      if (!this.realm.syncSession) {
+        throw new Error("No syncSession found on realm");
+      }
       await this.realm.syncSession.uploadAllLocalChanges();
```

Comment:
```suggestion
      if (this.realm.syncSession) {
        await this.realm.syncSession.uploadAllLocalChanges();
      }
```

Reply:
My thinking was that if the `syncSession` does not exist here, something has gone badly wrong, as we've created a realm with a sync config, so we should throw

Reply:
Ahh sorry, I hadn't expanded the code snippet. I agree, we're in a branch where sync is expected.